### PR TITLE
Add log to help debug 'unable to match..." cases

### PR DESCRIPTION
**Pull request by fernandog**
This helped find this wrong regex matching:

https://github.com/pymedusa/SickRage/pull/510

@duramato

### CODE REVIEW

File: sickbeard/name_parser/parser.py [hunk 1]
```diff
@@ -463,13 +463,13 @@ def parse(self, name, cache_result=True):
         final_result.quality = self._combine_results(file_name_result, dir_name_result, 'quality')
 
         if not final_result.show:
-            raise InvalidShowException("Unable to match {} to a show in your database".format
-                                       (name.encode(sickbeard.SYS_ENCODING, 'xmlcharrefreplace')))
+            raise InvalidShowException("Unable to match {0} to a show in your database. Parser result: {1}".format
+                                       (name.encode(sickbeard.SYS_ENCODING, 'xmlcharrefreplace'), file_name_result or dir_name_result))
```
Comment (labrys) [inline sickbeard/name_parser/parser.py]: Instead of encoding to sys encoding this should probably decode and be a unicode string (same with the other).

Comment (fernandog) [inline sickbeard/name_parser/parser.py]: This change should be in another PR right?
Because if if breaks we will revert the scope of this PR

Comment (labrys) [inline sickbeard/name_parser/parser.py]: No it should be another commit in this PR.  And then it would ofc require additional tests, if it fails then this commit would be reverted leaving the original commit.

Comment (fernandog) [inline sickbeard/name_parser/parser.py]: Until the PP encoding issues is really fixed we shoulnt change any encoding stuff.

This PR is done. Its just a informative PR

Comment (labrys) [inline sickbeard/name_parser/parser.py]: This encoding isn't related to the PP encoding issues.  The idea is it's safer to log unicode messages so making the string unicode and skipping the encode is less prone to errors because python itself does the unicode coercion.  e.g. 
```python
u"Unable to match {0} to a show in your database. Parser result: {1}".format(
    name, file_name_result or dir_name_result)
```

File: sickbeard/name_parser/parser.py [hunk 2]
```diff
@@ -463,13 +463,13 @@ def parse(self, name, cache_result=True):
         final_result.quality = self._combine_results(file_name_result, dir_name_result, 'quality')
 
         if not final_result.show:
-            raise InvalidShowException("Unable to match {0} to a show in your database. Parser result: {1}".format
-                                       (name.encode(sickbeard.SYS_ENCODING, 'xmlcharrefreplace'), file_name_result or dir_name_result))
+            raise InvalidShowException(u"Unable to match {0} to a show in your database. Parser result: {1}".format(
+                                         name, file_name_result or dir_name_result)
```
Comment (fernandog) [inline sickbeard/name_parser/parser.py]: @labrys

**Comment by fernandog** [comment]
@duramato any locale issues with that commit in pubdate2 branch?

**Comment by duramato** [comment]
all good

**Comment by duramato** [comment]
Approved
